### PR TITLE
Update coverage from MAVSDK tests

### PR DESCRIPTION
**Pull request by LorenzMeier**
This enables updating coverage information successfully from MAVSDK tests.

### CODE REVIEW

File: cmake/coverage.cmake [hunk 1]
```diff
@@ -34,16 +34,31 @@
 find_program(LCOV_PATH lcov)
 find_program(GENHTML_PATH genhtml)
 
+message(STATUS "Building for code coverage")
+
 # add code coverage build type
+if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang") OR ("${CMAKE_CXX_COMPILER_ID}" MATCHES "AppleClang"))
```
Comment (dagar) [inline cmake/coverage.cmake]:
```suggestion
if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
```

"Clang" MATCHES "Clang" and "AppleClang".

File: .github/workflows/sitl_tests_coverage.yml [hunk 2]
```diff
@@ -16,8 +16,20 @@ jobs:
...
       run: make tests_integration_coverage
-    - name: Upload coverage to Codecov
+    # - name: Download newer mavsdk
+    #   run: git clone https://github.com/mavlink/MAVSDK.git && cd MAVSDK && git submodule init && git submodule update
+    # - name: Build newer mavsdk
+    #   run: mkdir -p MAVSDK/build/default && cd MAVSDK/build/default && cmake ../.. && make -j4 && make install
+    - name: Run Coverage Tests
+      run: make tests_integration_coverage
+    - name: Upload coverage information to Coveralls
+      uses: coverallsapp/github-action@master
```
Comment (dagar) [inline .github/workflows/sitl_tests_coverage.yml]: Is coveralls an experiment or do you want both?

File: Makefile [hunk 3]
```diff
@@ -360,14 +360,16 @@ rostest: px4_sitl_default
...
 
 tests_integration_coverage:
 	@$(MAKE) clean
 	@$(MAKE) --no-print-directory px4_sitl_default PX4_CMAKE_BUILD_TYPE=Coverage
 	@$(MAKE) --no-print-directory px4_sitl_default sitl_gazebo
 	@$(MAKE) --no-print-directory px4_sitl_default mavsdk_tests
-	@"$(SRC_DIR)"/test/mavsdk_tests/mavsdk_test_runner.py --speed-factor 20 --iterations 1 --fail-early
+	@"$(SRC_DIR)"/test/mavsdk_tests/mavsdk_test_runner.py --speed-factor 15
+	@mkdir -p coverage
+	@lcov --directory build/px4_sitl_default --base-directory build/px4_sitl_default --gcov-tool gcov --capture -o coverage/lcov.info
```
Comment (dagar) [inline Makefile]: Manual lcov needed for Coveralls or a different reason?

Comment (LorenzMeier) [reply Makefile]: Actually for Codecov because its automatic parser gets the directories wrong and has no option that would fix it.

**Comment by dagar** [comment]
Rebased on master.

**Pull request closed without merging**